test/dotnet-publish.Tests/Microsoft.DotNet.Tools.Publish.Tests.cs

```diff
@@ -74,40 +97,25 @@ public void PublishOptionsTest(string framework, string runtime, string config,
         [Fact]
         public void ProjectWithContentsTest()
         {
-            // create unique directories in the 'temp' folder
-            var testDir = Temp.CreateDirectory();
-
-            var testAppDir = Path.Combine(_testProjectsRoot, "TestAppWithContents");
+            TestScenario scenario = TestScenario.Create(Path.Combine(_testProjectsRoot, "TestAppWithContents"), skipRestore: true);
+            TestInstance instance = scenario.CreateTestInstance()
+                                   .WithLockFiles()
+                                   .WithBinaries();
 
-            // copy projects to the temp dir
-            CopyProjectToTempDir(testAppDir, testDir);
-
-            // run publish
-            var testProject = GetProjectPath(testDir);
+            var testProject = Path.Combine(instance.TestRoot, "project.json");
```

Comment:
This kinda sucks too, we should look at the amount of `Path.Combine`s off the test root and see if there are common properties we should expose or helpers to make this easier.